### DISCUSSION TITLE
Assets automatically injected for non-livewire requests in tests

### DISCUSSION THREAD

**Discussion by hansnn** [livewire/livewire#6827] [7. Bugs (archived)]
### Livewire version

v3.0.3

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/hansnn/failing-livewire-test-case

### Steps To Reproduce

1. Create an HTTP test that calls an endpoint that returns a Livewire component.
2. Create an HTTP test that calls an endpoint that returns a non-livewire view.
3. In the 2nd test case, assert anything on the view (for example: `$response->assertViewHas('something')`)
4. Make sure that `inject_assets` is set to `true`
5. Run both tests at the same time, but make sure that the Livewire test runs before the other test.
6. The 2nd test will fail, because "the response is not a view".

### Additional information

If you run tests that are invoking both Livewire views and regular views at the same time, `inject_assets` is set to true, and the tests that invoke the Livewire views are run first: The tests for the regular view will contain the livewire assets, which causes some assertion methods on the view to fail.

A minimum reproducible is provided. You can check out [this commit](https://github.com/hansnn/failing-livewire-test-case/commit/fb05d525b2a85b877b32ca978dede7519295ecb4) specifically, which touches 3 files in a fresh Laravel installation with Livewire to introduce a failing test.
The failing test is as simple as this:
```php
class FailingTest extends TestCase
 {
     public function test_the_livewire_component_works()
     {
         $response = $this->get('/livewire-test');
         $response->assertSee('Hello World');
     }

     public function test_the_application_returns_a_successful_response(): void
     {
         $response = $this->get('/');
         $response->assertViewHas('a-property');
     }
 }
```
The 2nd test will fail with the error `The response is not a view.`. If you change the order of the tests, they will both pass.

The problem seems to be related to `SupportAutoInjectedAssets`, which sets `$hasRenderedAComponentThisRequest` to true when the `dehydrate` hook is invoked during the request to the Livewire component. The `SupportAutoInjectedAssets` is not cleaned up after the test, leaving the `$hasRenderedAComponentThisRequest` set to true during the 2nd test case, causing it to inject the assets when it shouldn't.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by rynomars** [comment]
This issue should be resolved in v3.4.0 with this PR #7758
